src/crystal/scheduler.cr

```diff
@@ -123,4 +138,68 @@ class Crystal::Scheduler
     @current.resume_event.add(0.seconds)
     resume(fiber)
   end
+
+  {% if flag?(:preview_mt) %}
+    @rr_target = 0
+
+    protected def find_target_thread
+      if @@workers.empty?
+        Thread.current
+      else
+        @rr_target += 1
+        @@workers[@rr_target % @@workers.size]
+      end
+    end
+
+    def run_loop
+      loop do
+        @lock.lock
+        if runnable = @runnables.shift?
+          @runnables << Fiber.current
+          @lock.unlock
+          runnable.resume
+        else
+          @sleeping = true
+          @lock.unlock
+          oid = @worker_out.read_bytes(UInt64)
+          fiber = Pointer(Fiber).new(oid).as(Fiber)
+
+          @lock.lock
+          @sleeping = false
+          @runnables << Fiber.current
+          @lock.unlock
+          fiber.resume
+        end
+      end
+    end
+
+    def send_fiber(fiber : Fiber)
+      @lock.lock
+      if @sleeping
+        @worker_in.write_bytes(fiber.object_id)
+      else
+        @runnables << fiber
+      end
+      @lock.unlock
+    end
+
+    @@workers = [] of Thread
+
+    def self.workers
+      @@workers
+    end
+
+    def self.init_workers
+      count = ENV["CRYSTAL_WORKERS"]?.try &.to_i || 4
```

Comment:
```suggestion
      count = ENV["CRYSTAL_WORKERS"]?.try &.to_i? || 4
```

If a non-integer like "foo" is passed to `CRYSTAL_WORKERS` this will crash the program. Maybe it's fine.

How do we want to handle it? I just checked in Go, they seem to completely ignore the value if it's not an integer. In our case it might be good to raise or abort the program, it might be a programmer mistake and we should detect that.

Reply:
I imagine there are a few cases you could cause this to fail.  I would be curious what negative numbers would do.  What about white listing the numbers?

```suggestion
      count = ("1".."100").include?(ENV["CRYSTAL_WORKERS"]?.try &.to_i) || 4
```

I would rather a raise an error for bad input rather to a logic error being risen from the results due to bad inputs.

Reply:
Why limit the number of cores? Cavium ThunderX (armv8, fully supported by Crystal) can have upto 128 cores for example. We don't know how many cores may be available tomorrow (or even today).

Thought coordinating that many cores without too much  will be huge challenge.